src/SFA.DAS.EmployerFinance/Models/LevyDeclarationSaga.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.EmployerFinance.Messages.Events;
+
+namespace SFA.DAS.EmployerFinance.Models
+{
+    public class LevyDeclarationSaga : Entity
+    {
+        public int Id { get; private set; }
+        public LevyDeclarationSagaType Type { get; private set; }
+        public DateTime PayrollPeriod { get; private set; }
+        public long HighWaterMarkId { get; private set; }
+        public DateTime Created { get; private set; }
+        public DateTime? Updated { get; private set; }
+        public int ImportPayeSchemeLevyDeclarationsTasksCount { get; private set; }
+        public int ImportPayeSchemeLevyDeclarationsTasksCompleteCount { get; private set; }
+        public DateTime? ImportPayeSchemeLevyDeclarationsTasksFinished { get; private set; }
+        public int UpdateAccountTransactionBalancesTasksCount { get; private set; }
+        public int UpdateAccountTransactionBalancesTasksCompleteCount { get; private set; }
+        public DateTime? UpdateAccountTransactionBalancesTasksFinished { get; private set; }
+        public bool IsComplete { get; private set; }
+        public TimeSpan Timeout => TimeSpan.FromMinutes(2);
+
+        public LevyDeclarationSaga(DateTime payrollPeriod, IReadOnlyCollection<AccountPayeScheme> accountPayeSchemes)
+        {
+            Type = LevyDeclarationSagaType.All;
+            PayrollPeriod = payrollPeriod;
+            Created = DateTime.UtcNow;
+            HighWaterMarkId = accountPayeSchemes.OrderByDescending(aps => aps.Id).Select(aps => aps.Id).First();
+            ImportPayeSchemeLevyDeclarationsTasksCount = accountPayeSchemes.Select(aps => aps.EmployerReferenceNumber).Distinct().Count();
```

Comment:
When calculating the `ImportPayeSchemeLevyDeclarationsTasksCount` here, we take the count of distinct `EmployerReferenceNumber`s, but in `ImportLevyDeclarationsCommandHandler.Handle()`, we `Take(saga.ImportPayeSchemeLevyDeclarationsTasksCount)` on the non-distinct set. If there are any duplicates, does that mean some PayeSchemes won't be imported?

Reply:
I've removed the `Distinct()` calls and added `builder.HasQueryFilter(aps => aps.Deleted == null);` to `AccountPayeSchemeConfiguration` :+1: